### DISCUSSION TITLE
Existing Framework Required for Web Components and HTML?

### DISCUSSION THREAD

**Discussion by davidhmays** [storybookjs/storybook#24788] [Help]
I'm interested in using storybook for the ability to create and test components in isolation.
I'm confused as to why the docs for both Web Components and HTML both say the first thing you run is storybook init, but that,  

> "storybook init is not made for empty projects
> Storybook needs to be installed into a project that is already set up with a framework. It will not work on an empty project".

If I'm designing components in isolation with Web Components or HTML, there _is no framework_? What is the minimum I need in the existing project to use with these?

[HTML Docs](https://storybook.js.org/docs/html/get-started/install/)
[Web Components Docs](https://storybook.js.org/docs/web-components/get-started/install)

**Comment by break-stuff** [comment]
If it can't detect the project type, it will usually ask you what version of storybook you want to use.

**Reply by break-stuff** [reply]
If you're creating web components, I recommend the web components version since there is some additional tooling set up in there to make integration easier.

**Comment by zz-james** [comment]
Hello, I am also working on a project which has no framework, vanilla web components, no lit. 

when I try to install the installer asks me for a project template and the choices are all frameworks + typescript, does it matter if I choose one and then not use it? or is there an optional param to send to the install command.

I have seen a lot of documentation online but it all seems out of date now (October 2025)

Thanks for any help

**Comment by break-stuff** [comment]
@zz-james No, not really. Just be sure to choose `web_components` or a web component-related option like Lit. The implementation is very framework agnostic.

**Reply by zz-james** [reply]
Thanks I got it now, original when I was running the npx command I wasn't seeing the installation option that I needed but after I npm init the folder the installer gave me more options, everything plain sailing since then 👍
